src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```diff
@@ -544,15 +547,158 @@ public boolean isPluginEnabled(Plugin plugin)
 	public Plugin loadPlugin(File file)
 			throws InvalidPluginException, InvalidDescriptionException, UnknownDependencyException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if (!file.getName().endsWith(".jar")) 
+		{
+			throw new InvalidPluginException("Plugin must be a .jar file");
+		}
+
+		try 
+		{
+			File tempDir = createTemporaryDirectory("MockBukkit-plugin-" + file.getName().replace(".", "-"));
+			Path targetPath = Files.copy(file.toPath(), tempDir.toPath().resolve(file.getName()), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Why is the Jar file to be loaded copied to a temporary directory? Is it not possible to load it directly from it's original location?

Reply:
I wanted to recreate the already existing loading mechanism of the "internal" plugins as close as possible. As they are passed a temporary data directory I thought of recreating that here. It would be no problem though to change that and load the plugin directly in that folder.

Reply:
We do indeed need a temporary data directory in case the plugin wants to store some configuration or some data. However, we do not have to copy the jar file itself to this directory.

We also have to make sure to remove that data after the test has finished in order to prevent someone's disk from filling up if they have to run hundreds of tests. On my Linux system, for instance, it's all being stored in `/tmp` which is just a ram disk. It's really fast and won't wear out my SSD, but the amount of data I can store there is limited by how much memory I have. Running a test suite consisting of many tests should not be filling that up.

The current implementation has a `List<File> temporaryFiles` which contains all temporary directories and files that were created during a single unit test. When `MockBukkit.unload()` is called, each directory and file in this list gets removed in order to avoid this issue.